### PR TITLE
Override endpoints hook

### PR DESCRIPTION
**Pull request by markusjura**

### CODE REVIEW

File: src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala [hunk 1]
```diff
@@ -16,7 +16,8 @@ object Import {
 
   /**
    * Represents a service endpoint.
-   * @param bindProtocol the protocol to bind for this endpoint, e.g. "http"
+    *
+    * @param bindProtocol the protocol to bind for this endpoint, e.g. "http"
```
Comment (huntc) [inline src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala]: Alignment

File: src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala [hunk 2]
```diff
@@ -149,10 +150,10 @@ object SbtBundle extends AutoPlugin {
   val autoImport = Import
 
   private final val Sha256 = "SHA-256"
+  private final val Utf8 = Charset.forName("utf-8")
+  private[bundle] final val DefaultEndpoints = Map("web" -> Endpoint("http", 0, Set(URI("http://:9000"))))
```
Comment (huntc) [inline src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala]: I think we should make this public and place it in the companion obj. Wdyt?

File: src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala [hunk 3]
```diff
@@ -245,11 +246,14 @@ object SbtBundle extends AutoPlugin {
   private val systemConfigName = taskKey[(String, Option[String])]("")
   private val systemVersionConfigName = taskKey[(String, Option[String])]("")
 
+  private[bundle] val overrideEndpoints = taskKey[Map[String, Endpoint]]("")
```
Comment (huntc) [inline src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala]: Let's make this public too

**Comment by markusjura** [comment]
Provides a new public task key `overrideEndpoints` to define the endpoints programmatically. This new key is used by `sbt-lagom-bundle` in order to resolve the Lagom endpoints in the descriptor. It is necessary to create a the new task key `overrideEndpoints` instead of changing the `endpoints` settings key to a task key. The reason in Lagom we either:
- Take the `BundleKeys.endpoints` if the user has specified any
- Otherwise take the acls defined in the service descriptor

This logic can only achieved by introducing this new task key `overrideEndpoints`.

**Pull request merged**